File: LeetCode/150.Evaluate_Reverse_Polish_Notation/150.Evaluate_Reverse_Polish_Notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();

        for (String str : tokens) {
            switch (str) {
                case "+":
                    int a = stack.pop(), b = stack.pop();
                    stack.push(a + b);
                    break;
                case "-":
                    int c = stack.pop(), d = stack.pop();
                    stack.push(d - c);
                    break;
                case "/":
                    int e = stack.pop(), f = stack.pop();
                    stack.push(f / e);
                    break;
                case "*":
                    int g = stack.pop(), h = stack.pop();
                    stack.push(g * h);
                    break;
                default:
                    stack.push(Integer.parseInt(str));
            }
        }

        return stack.pop();
    }
}

File: LeetCode/150.Evaluate_Reverse_Polish_Notation/150.Evaluate_Reverse_Polish_Notation.py
#!/usr/bin/env python3


class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for t in tokens:
            if t in set(["+", "-", "*", "/"]):
                last = stack.pop()
                if t == "+":
                    stack[-1] += last
                elif t == "-":
                    stack[-1] -= last
                elif t == "*":
                    stack[-1] *= last
                elif t == "/":
                    a = stack[-1]
                    stack[-1] = int(a / last)
            else:
                stack.append(int(t))

        return stack[-1]
